launch_testing/launch_testing_examples/launch_testing_examples/record_rosbag_launch_test.py

```diff
@@ -0,0 +1,83 @@
+# Copyright 2021 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import os
+import shutil
+import tempfile
+import time
+import unittest
+
+import launch
+import launch.actions
+import launch_ros.actions
+import launch_testing.actions
+import launch_testing.markers
+import pytest
+
+import yaml
+
+
+@pytest.mark.launch_test
+@launch_testing.markers.keep_alive
+def generate_test_description():
+    rosbag_dir = os.path.join(tempfile.mkdtemp(), 'test_bag')
+
+    node_list = [
+        launch_ros.actions.Node(
+            executable='talker',
+            package='demo_nodes_cpp',
+            name='demo_node'
+        ),
+        launch.actions.ExecuteProcess(
+            cmd=['ros2', 'bag', 'record', '-a', '-o', rosbag_dir],
+            output='screen'
+        ),
+        launch_testing.actions.ReadyToTest()
+    ]
+
+    return launch.LaunchDescription(node_list), {'rosbag_dir': rosbag_dir}
+
+
+class DelayShutdown(unittest.TestCase):
+
+    def test_delay(self):
+        """Delay the shutdown of processes so that rosbag can record some messages."""
+        time.sleep(3)
+
+
+# TODO : Test fails on windows, to be fixed
```

Comment:
Please create at ticket to track re-enabling this test on Windows.

Reply:
Done : https://github.com/ros2/rosbag2/issues/926